### PR TITLE
feat: add platform-specific configuration protos to `kfp-pipeline-spec` package

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Adds platform-specific wrapper protos to `kfp-pipeline-spec` package.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request 
title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: api/v2alpha1/pipeline_spec.proto [hunk 1]
```diff
@@ -998,3 +998,21 @@ message PipelineStateEnum {
...
+}
+
+message SinglePlatformSpec {
+  // Mirrors PipelineSpec.deployment_spec structure
+  PlatformDeploymentConfig deployment_spec = 1;
+}
+
+
+message PlatformDeploymentConfig {
+  // Map of executor label to task-level config
```
Comment (chensun) [inline api/v2alpha1/pipeline_spec.proto:1015]: nit: should we say `executor-level` to be more accurate?

**Review by chensun — COMMENTED** [review]
/lgtm

One nitpick

File: api/v2alpha1/pipeline_spec.proto (hunk 1 above)
Comment (connor-mccarthy) [reply api/v2alpha1/pipeline_spec.proto]: Yes, good catch. Updated.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

**Pull request merged**